crates/shared/src/oneinch_api.rs

```diff
@@ -384,6 +386,55 @@ where
     serde_json::from_str(&response?).context("1inch result parsing failed")
 }
 
+#[derive(Debug, Clone)]
+pub struct ProtocolCache(Arc<Mutex<TimedSizedCache<Vec<String>, Vec<String>>>>);
```

Comment:
I think we should cache the raw list of protocols that 1nch returns instead of this construct where the entries in the cache depend on the disabled protocols. The underlying data is always the same so I imagine there would be a "get current protocols" function that either returns the cached value or updates the cache and returns the value.
That you might or might not implement with this cached trait. I'm saying might not because when there is only one value it might be so simple that the crate doesn't add anything.

Reply:
One advantage of caching the protocol list instead of the enabled/disabled protocols is that it, in theory, allows multiple instances of `1Inch{PriceEstimator,Solver}` to share the same cache.

That being said, we don't really do that now so maybe its a bit of "overarchitecting" on my part.

> there is only one value it might be so simple that the crate doesn't add anything.

> But even if the general use case is only storing a single value, I would prefer using the crate over implementing a cache myself.

I don't have a strong opinion here. If we added `cached` as a dependency _just_ for this cache, I would tend to agree that we shouldn't use, but since its already included in our dependency tree - why not?

Reply:
You could use `TimedCache<(), Vec<String>>` if you wanted to keep a single list right?

Reply:
> I wanted to skip filtering out disabled protocols every time we need the allowed list

I like doing it that way a lot more. Filtering takes no time compared to making the network request and it has the advantage of simpler code because only one thing is cached.